tests/cpgrid/addLgrsOnDistributedGrid_test.cpp

```diff
@@ -497,9 +519,21 @@ BOOST_AUTO_TEST_CASE(threeLgrs)
         // LGR3 element indices = 183 -> refined into 4x4x4 = 64 cells                LGR3 dim 4x4x4
         grid.addLgrsUpdateLeafView(cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
 
-        // 10x8x8 -6 + 32 + 27 + 64 = 757
-        // 11x9x9 + (75-18) + (64-8) + (125-8) = 1121
+        // Leaf grid view total amount of cells: 10x8x8 -6 + 32 + 27 + 64 = 757
         refinePatch_and_check(grid, cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
+
+        // Check global id is not duplicated for points
+        std::vector<int> localPointIds_vec;
+        localPointIds_vec.reserve(grid.currentData().back()->size(3));
+        for (const auto& point : vertices(grid.leafGridView())) {
+            // Notice that all partition type points are pushed back. Selecting only interior points does not bring us to the expected value.
+            localPointIds_vec.push_back(grid.currentData().back()->globalIdSet().id(point));
+        }
+        auto [allGlobalIds_points, displPoint ] = Opm::allGatherv(localPointIds_vec, grid.comm());
+        const std::set<int> allGlobalIds_points_set(allGlobalIds_points.begin(), allGlobalIds_points.end());
+
+        // Leaf grid grid total ampunt of points: 11x9x9 + (75-18) + (64-8) + (125-8) = 1121
+        BOOST_CHECK( allGlobalIds_points_set.size() == 1121 );
```

Comment:
This is not of much use here as global ids for points are part of the next step.  Maybe remove it for now?

Reply:
This is one of the test cases for fully interior LGRs, and I’d like to keep it. The reason this check appears here now is that the previous check for non-duplicated global ids for points in fully interior LGRs was removed. That old check relied on manual calculations using parent cell blocks and LGR dimensions based on ijk-computations, rather than a more stable, long-term solution using DUNE’s grid interface.

I promise to clean this up as soon as the implementation for unique global ids for points is integrated!

Reply:
Good, then let's not forget.